### DISCUSSION TITLE
error when executing report to Azure and all analysis

### DISCUSSION THREAD

**Discussion by vanessav5** [silverhack/monkey365#108] [General]
Hi,

I need your help, I'd like to execute monkey365 to obtain all Azure reports.  I executed this:

Invoke-Monkey365 -PromptBehavior SelectAccount -Instance Azure -Analysis All -TenantID XXX-XXX-XXX -ExportTo HTML

But obtain the error:

Write-Error : [12:37:56:792] - [Invoke-AzureScanner] - No se puede convertir el argumento "collection", con el valor: "@{id=/subscriptions/XXXXXX/resourceGroups/RG_SENTINEL_WORKSPACE/providers/Microsoft.OperationalInsights/workspaces/XXXx-Sentinel;
name=XXX-Sentinel; type=Microsoft.OperationalInsights/workspaces; location=westeurope; tags=}", para "AddRange", al tipo
"System.Collections.Generic.IEnumerable`1[System.Object]": "No se puede convertir el valor "@{id=/subscriptions/XXXXXXXXXXXXXXXXXXXXXX/
resourceGroups/RG_SENTINEL_WORKSPACE/providers/Microsoft.OperationalInsights/workspaces/XXXX-Sentinel; name=XXXX-Sentinel;
type=Microsoft.OperationalInsights/workspaces; location=westeurope; tags=}" de tipo "System.Management.Automation.PSCustomObject" al tipo
"System.Collections.Generic.IEnumerable`1[System.Object]".". LineNumber: En C:\monkey365\core\analysis\Invoke-AzureScanner.ps1: 97 Carácter: 49
+ ... resources = Get-MonkeyAzResource -ResourceGroupNames $rg_names -Diagn ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ - exception - WST-2129.dc.xxxxx.com -
En C:\monkey365\core\analysis\Invoke-AzureScanner.ps1: 136 Carácter: 9
+         Write-Error $_
+         ~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Write-Error

Could you help me, please?

Thank you so much for your attention.

Regards

Vanessa

**Comment by silverhack** [comment]
Hi @vanessav5, 

Seems that you have **only one** resource group (not a list of objects) and monkey raises an exception, so it's a bug. I just have updated the [dev branch](https://github.com/silverhack/monkey365/tree/develop), and this should be now resolved. 

Could you please try and do let me know?
Gracias,

**Reply by vanessav5** [reply]
Hi,

I've downloaded the dev version but I obtain this error:

PS C:\monkey365-develop> Invoke-Monkey365 -PromptBehavior SelectAccount -Instance Azure -Analysis All -TenantID XXXXX -ExportTo HTML
[ParameterError] Unable to create Monkey365 object: Excepción al llamar a ".ctor" con los argumentos "1": "No se admite el formato proporcionado de la ruta de acceso."
En C:\monkey365-develop\core\init\New-O365Object.ps1: 240 Carácter: 13
+             throw ("[ParameterError] {0}: {1}" -f "Unable to create M ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: ([ParameterError...uta de acceso.":String) [], RuntimeException
    + FullyQualifiedErrorId : [ParameterError] Unable to create Monkey365 object: Excepción al llamar a ".ctor" con los argumentos "1": "No se admite el formato proporcionado de
   la ruta de acceso."


I'll be waiting for your answer.

Regards,


VAnessa

**Comment by silverhack** [comment]
Hi @vanessav5, 

This is because you are trying to import the PowerShell module using an incorrect path. You should rename the folder `monkey365-develop` to `monkey365` and then import it in a new console session. 

If for example the module is located in c:\, you should do the following:

`Import-Module C:\monkey365`

Could you please try and do let me know? 

Gracias,

**Reply by vanessav5** [reply]
Hi @silverhack 

Perfect, It's working.

Thank you for your support.

Vanessa
